canonicalwebteam/launchpad/models.py

```diff
@@ -64,18 +68,70 @@ def __init__(self, username, token, secret, session):
             f'oauth_signature="&{secret}"'
         )
 
-    def _request(self, path, method="GET", params={}, data={}):
+    def request(self, url, method="get", params=None, data=None, json=None):
         """
-        Makes a raw HTTP request and returns the response.
+        Make a request through the configured API session
         """
 
-        url = f"https://api.launchpad.net/devel/{path}"
-
-        response = self.session.request(method, url, params=params, data=data)
+        response = self.session.request(
+            url=url, method=method, params=params, data=data, json=json
+        )
         response.raise_for_status()
 
         return response
 
+    def get_collection_entries(self, url, params=None):
+        """
+        Return collection items from the API
+        """
+
+        return self.request(url, params=params).json().get("entries", [])
+
+    def create_system_build_webhook(self, system, delivery_url):
+        """
+        Create a webhook for the given system to trigger when a
+        build is created or updates, if it doesn't exist already.
+        If it exists, raise WebhookExistsError
+        """
+
+        system_year = re.match(r"^[^\d]+(?:64)?(\d{2})(\.\d{2})?$", system)[1]
+        codename = self.system_codenames[system_year]
+        project = "ubuntu-core"
+        if system.startswith("classic"):
+            project = "ubuntu-cpc"
+
+        webhooks = self.get_collection_entries(
+            "https://api.launchpad.net/devel/"
+            f"~{self.username.replace('.', '')}/"
```

Comment:
@nottrobin just curiosity, why the replace?

Reply:
Launchpad seems to need it. No idea why.

Reply:
The username that we use in Snapcraft is `build.snapcraft.io` and I didn't have any issue with the dots and the username that you use is `imagebuild` without any dots, right?

https://launchpad.net/~imagebuild

Reply:
I have no idea. But for some reason, the name here is `imagebuild` and yet when we log in to the API we have to use the form `image.build`. Maybe it's something about this user specifically then? In which case I suppose I'd need to add an extra parameter, and this conditional is wrong. Worth digging into. Not sure I have time right now.

Reply:
No worries, interesting, my concern is that we need our user with the dots in snapcraft but if the test is passing I guess you are doing the replace only for your endpoints.